### DISCUSSION TITLE
NER model training: relation between `E` and `#`

### DISCUSSION THREAD

**Discussion by DavidGerva** [explosion/spaCy#8394] [Help: Other Questions] [answered]
I'm training a NER model to recognise very specific entities in a domain and I cannot understand what is the meaning of some output of the train procedure:
![image](https://user-images.githubusercontent.com/1856249/122014098-1e8df700-cdbf-11eb-8f53-9d69dc0aeb5c.png)

`E` are the epochs but what about `#`?
Is `#` the number of data processed in the given epoch?

And, related to this, in the spacy `config.cfg` file I can set both the params `patience` and `max_steps` and the only documentation I could find is this one: `max_steps and/or patience are used to determine when the training should stop`.
I did some tests and `patience` has never been considered, always on only `max_steps`.
Which is the difference between the two?

**Comment by polm** [comment, marked as answer]
Please see https://github.com/explosion/spaCy/discussions/7731 and https://github.com/explosion/spaCy/discussions/7465. 

Basically # is iterations/steps and E is epochs.

`max_steps` is an absolute limit, but if you go `patience` steps without improvement training will terminate.

**Reply by DavidGerva** [reply]
Thank you again @polm, much clearer now!
